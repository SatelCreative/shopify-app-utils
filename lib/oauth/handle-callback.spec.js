const handleCallback = require('./handle-callback');

describe('Handle Callback', () => {
  it('should return token when request is valid', async () => {
    // const response = await handleCallback({
    //   url: '/?shop=example.myshopify.com&state=123',
    //   nonce: '123',
    //   apiKey: '123',
    //   secret: '123',
    // });
    throw new Error('TODO');
  });
});
