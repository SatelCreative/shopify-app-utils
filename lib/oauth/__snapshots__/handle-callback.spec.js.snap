// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle Callback should fail on invalid nonce 1`] = `
Array [
  Object {
    "nonce": "my-state",
    "shop": "example.myshopify.com",
  },
]
`;

exports[`Handle Callback should return token when request is valid 1`] = `
Array [
  Object {
    "client_id": "api-key",
    "client_secret": "hunter2",
    "code": "my-code",
  },
]
`;

exports[`Handle Callback should return token when request is valid 2`] = `
Array [
  Object {
    "nonce": "my-state",
    "shop": "example.myshopify.com",
  },
]
`;

exports[`Handle Callback should return token when request is valid 3`] = `
Array [
  Object {
    "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZXMiOlsicmVhZF9wcm9kdWN0cyJdLCJzaG9wIjoiZXhhbXBsZS5teXNob3BpZnkuY29tIiwiaWF0IjoxNTMzMTYwODAwfQ.jIeMf0i_zhj31Byb-FCq68T5TRfVqAOfkBTEFAL0c18",
    "scopes": Array [
      "read_products",
    ],
    "shop": "example.myshopify.com",
    "token": "access_token",
  },
]
`;
